File: patterns/debounce/rules-js/debounce-simple.js
let timer = null;

rules.JSRule({
  name: "Debounce Person1",
  triggers: [triggers.ItemStateChangeTrigger("Person1PresenceSensor")],
  execute: (event) => {
    if (timer !== null) timer.cancel();

    const sensorState = items.getItem("Person1PresenceSensor").state;
    const delaySeconds = sensorState === "ON" ? 0 : 120;

    timer = actions.ScriptExecution.createTimer(
      time.ZonedDateTime.now().plusSeconds(delaySeconds),
      () => {
        if (items.getItem("Person1Presence").state !== sensorState) {
          items.getItem("Person1Presence").postUpdate(sensorState);
          console.log("Person1Presence uebernimmt " + sensorState + " (Verzoegerung=" + delaySeconds + "s)");
        }
        timer = null;
      }
    );
  }
});

File: patterns/debounce/rules-python/debounce_simple.py
import threading

from openhab import rule, Registry, logger
from openhab.triggers import ItemStateChangeTrigger

timer = None


@rule(triggers=[ItemStateChangeTrigger("Person1PresenceSensor")])
class DebouncePerson1:
    def execute(self, module, input):
        global timer
        if timer is not None:
            timer.cancel()

        sensor_state = str(Registry.getItem("Person1PresenceSensor").getState())
        delay_seconds = 0 if sensor_state == "ON" else 120

        def on_expire():
            global timer
            proxy = Registry.getItem("Person1Presence")
            if str(proxy.getState()) != sensor_state:
                proxy.postUpdate(sensor_state)
                logger.info(
                    "Person1Presence uebernimmt {} (Verzoegerung={}s)".format(sensor_state, delay_seconds)
                )
            timer = None

        timer = threading.Timer(delay_seconds, on_expire)
        timer.start()
